pypeit/flatfield.py

```diff
@@ -448,7 +447,30 @@ def run(self, debug=False, show=False):
         """
         # Fit it
         # NOTE: Tilts do not change and self.slits is updated internally.
-        self.fit(spat_illum_only=self.spat_illum_only, debug=debug)
+        if not self.flatpar['flatfield_structure']:
```

Comment:
Please provide some inline comments about what flatfield_structure is doing to document this functionality. Maybe also point out that it is IFU specific.